caffeinated/admin/new/pricing/templates/event_tickets_metabox_main.template.php

```diff
@@ -140,15 +140,15 @@ class="ee-icon ee-icon-tickets ee-icon-size-20"></span><?php
                         'Available Tickets',
                         'event_espresso'
                     ); ?></span></h3>
-        <div class="event-tickets-container ee-create-ticket-button"<?php echo $show_tickets_container; ?>>
+        <div class="event-tickets-container ee-create-ticket-button">
```

Comment:
i'm assuming (and hoping) this template variable doesn't exist anywhere?

Reply:
It was always empty so I deleted it instead of escaping an empty string 😉

Reply:
was the variable being passed though?
it may have been run through a filter so that users could change it.
so it may have always been empty using the plugin as is, but may have a value when filtered on a user site.
can you plz confirm?

Reply:
It defined in `espresso_events_Pricing_Hooks::pricing_metabox` function located in `caffeinated/admin/new/pricing/espresso_events_Pricing_Hooks.class.php` file.

In that function there is no filter to modify the values of template args but there is a general hook in `EEH_Template::display_template` function named `FHEE__EEH_Template__display_template__template_args` which is used for all templates and not specifically this one.

Reply:
hmmm... ya... we should keep that in just in case. 
Can you plz add the `$show_tickets_container` variable back and escape it?
thnx

Reply:
you'll have to use `wp_kses()` with your custom form attributes

Reply:
> After applying wp_kses function we will have exact same string.

??? 

How do you figure that? 
You're saying it wouldn't strip out anything that isn't an allowed tag or attribute?
I would expect the resulting string to only be `value="foobar"` since `value` is likely the only attribute that is currently allowed. The others would be removed, no?

IIRC we filtered the allowed tags, so if a user wants to inject `never-exists-such-attribute` they will need to first add that to the allowed attributes. If we didn't filter the allowed tags then we should.

Am i missing something here?

Reply:
$show_tickets_container = 'value="foobar" notallowed="value" never-exists-such-attribute="yes" onerror="alert(\'XSS\');"';
    <div class="event-tickets-container ee-create-ticket-button" <?php echo wp_kses($show_tickets_container, AllowedTags::getAllowedTags()); ?>">

Above code will return the exact string becuase `wp_kses` function won't consider the string as HTML (which is not) and returns it without any change. If we put a HTML code then it works as expected and will remove the unwanted attributes and tags.

Another option can be to convert whole `div` (from opening tag to the closing one) to a PHP string and then apply `wp_kses` to the string and print it but it will very messy.

Reply:
didn't know `wp_kses()` had to be "primed" with a tag to make it work! TIL

hmm ok... so what about just slapping `sanitize_text_field()` on that then?
That won't work on full HTML but might be ok in this case